### PR TITLE
Handle permission for babelfish internal procedures 

### PR DESCRIPTION
**Pull request by tanscorpio7**
### Description

Revoke permissions for babelfish internal procedures from public. 

Also push fix for -- terminate-psql-conn in JDBC test frameworks

### Issues Resolved

[BABEL-4503]

### Sign Off

Signed-off-by: Tanzeel Khan <tzlkhan@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql [hunk 1]
```diff
@@ -4267,3 +4267,16 @@ END
...
+REVOKE EXECUTE ON PROCEDURE sys.babel_coercion_initializer FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_catalog_initializer FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_create_builtin_dbs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_drop_all_dbs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_initialize_logins FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.babel_drop_all_logins FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.initialize_babel_extras FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.analyze_babelfish_catalogs FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.initialize_babelfish FROM PUBLIC;
+REVOKE EXECUTE ON PROCEDURE sys.remove_babelfish FROM PUBLIC;
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4281]: After this change only the function owner will have execute rights on it, right? Just thinking if we can add positive test case for them somehow. At least, you can test the positive cases locally and add it in the description.

Comment (HarshLunagariya) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4281]: Otherwise changes looks good.

Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4281]: Yes, will test it locally.
But all these procedures are executed while installing or removing babelfish (directly or indirectly). That could be considered as a tests. Right ?

Comment (HarshLunagariya) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4281]: Yes.

Comment (surendravishnoi) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:4281]: we should add tests where 1) Existing login/user which can execute these procedures 2) Create a new login/user which can execute these procedures 3) Corresponding PSQL behavior 4) Additional tests to cover Gitfarm side of differences

**Pull request closed without merging**